pkg/cloud/cloud_io.go

```diff
@@ -308,7 +308,7 @@ func BackgroundPipe(
 ) io.WriteCloser {
 	pr, pw := io.Pipe()
 	w := &backgroundPipe{w: pw, grp: ctxgroup.WithContext(ctx), ctx: ctx}
-	w.grp.GoCtx(func(ctc context.Context) error {
```

Comment:
wanted to confirm my understanding: b/c the context variable defined in the function was mistyped as `ctc`, this closure accidentally captured the context passed to `BackgroundPipe()`, instead of using the ctxgroup's context. 

What's subtle here is that the ctx passed to the ctxgroup is the same as the one this function accidentally captures. So I don't quite undestand why this typo threw everything out of wack. Perhaps something to do with variable copying..

Reply:
It doesn't really manifest as a bug, I was chasing some weird behavior in the hierarchy of spans on tracez and I noticed this. Your understanding is the same as mine!